File: featurea/src/main/java/nyc/friendlyrobot/featurea/ActivityComponent.java
package nyc.friendlyrobot.featurea;

import android.app.Activity;

import dagger.BindsInstance;
import dagger.Component;
import nyc.friendlyrobot.base.ActivityScoped;
import nyc.friendlyrobot.base.BaseComponent;

@ActivityScoped
@Component(dependencies = BaseComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    void inject(FeatureAActivity mainActivity);

    @Component.Builder
    interface Builder {
        ActivityComponent build();
        Builder appComponent(BaseComponent baseComponent);

        @BindsInstance
        Builder activity(Activity activity);
    }

}

File: app/src/main/java/nyc/friendlyrobot/featuredagger/ActivityComponent.kt
package nyc.friendlyrobot.featuredagger

import dagger.Module
import dagger.Subcomponent
import nyc.friendlyrobot.base.ActivityScoped

@Subcomponent(modules = arrayOf(ActivityModule::class))
@ActivityScoped
interface ActivityComponent {
    fun inject(activity: BaseActivity)
}

@Module
class ActivityModule {

}
